Add xsrf token header if cookie is present (#9471)

tepi opened the pull request:
Adds custom X-XSRF-TOKEN header to requests, if XSRF-TOKEN cookie is set

Fixes #9471

**Check when you have completed**
[ ] Valid tests for the pull request
[x] Contributing guidelines implemented

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/10953)
<!-- Reviewable:end -->

tepi committed:
Add xsrf token header if cookie is present (#9471)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/framework?pullRequest=10953) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/framework?pullRequest=10953) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/framework?pullRequest=10953) it.</sub>

elmot commented:
Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/vaadin/framework/10953#-LDpjQZL7Vw6eoCdpSWd:-LDpjQZL7Vw6eoCdpSWe:b-6dtty6) (no related file):*
I have a feeling this change is actually an application security degradation. Having XSRF value in cookies does not look safe to me. What do you think, @legioth ?

---

*[server/src/main/resources/VAADIN/vaadinBootstrap.js, line 39 at r2](https://reviewable.io/reviews/vaadin/framework/10953#-LDpivob763RSX6dw2M6:-LDpivob763RSX6dw2M7:b-zfpxsa) ([raw file](https://github.com/vaadin/framework/blob/778e751a5bfa4367aefd03fd6ce3262f3332b4d0/server/src/main/resources/VAADIN/vaadinBootstrap.js#L39)):*
> ```JavaScript
>         }
>     };
>     
> ```

extra spaces
please run `mvn process-source` or `mvn install` before commiting

---

*[server/src/main/resources/VAADIN/vaadinBootstrap.js, line 203 at r2](https://reviewable.io/reviews/vaadin/framework/10953#-LDpj8xZ6x3EwIEYgzjU:-LDpj8xZ6x3EwIEYgzjV:b-k10qpq) ([raw file](https://github.com/vaadin/framework/blob/778e751a5bfa4367aefd03fd6ce3262f3332b4d0/server/src/main/resources/VAADIN/vaadinBootstrap.js#L203)):*
> ```JavaScript
>                 // send parameters as POST data
>                 r.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
>                 
> ```

extra spaces

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/framework/10953)*
<!-- Sent from Reviewable.io -->

tepi commented:
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/vaadin/framework/10953#-LDpjQZL7Vw6eoCdpSWd:-LDq9qbFDQwXfVKROQxw:bjmqas0) (no related file):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

I have a feeling this change is actually an application security degradation. Having XSRF value in cookies does not look safe to me. What do you think, @legioth ?
</blockquote></details>

This change does nothing to application security as-is. It just checks for the value in the cookie and adds it to the custom header. This enables use cases where the server is adding this cookie and expects the header to be present in all requests.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/framework/10953)*
<!-- Sent from Reviewable.io -->

tepi commented:
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[server/src/main/resources/VAADIN/vaadinBootstrap.js, line 39 at r2](https://reviewable.io/reviews/vaadin/framework/10953#-LDpivob763RSX6dw2M6:-LDqAg3z3C5VBskVoDqD:b-31gcu0) ([raw file](https://github.com/vaadin/framework/blob/778e751a5bfa4367aefd03fd6ce3262f3332b4d0/server/src/main/resources/VAADIN/vaadinBootstrap.js#L39)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

extra spaces
please run `mvn process-source` or `mvn install` before commiting
</blockquote></details>

I run both, it did no changes to this file (or the other one you commented on). What exactly is wrong?

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/framework/10953)*
<!-- Sent from Reviewable.io -->

tepi commented:
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[server/src/main/resources/VAADIN/vaadinBootstrap.js, line 39 at r2](https://reviewable.io/reviews/vaadin/framework/10953#-LDpivob763RSX6dw2M6:-LDqEmiU11PqOQUH1crv:b-rswj4a) ([raw file](https://github.com/vaadin/framework/blob/778e751a5bfa4367aefd03fd6ce3262f3332b4d0/server/src/main/resources/VAADIN/vaadinBootstrap.js#L39)):*
<details><summary><i>Previously, tepi (Teppo Kurki) wrote…</i></summary><blockquote>

I run both, it did no changes to this file (or the other one you commented on). What exactly is wrong?
</blockquote></details>

Never mind, found the issue and fixed.

---

*[server/src/main/resources/VAADIN/vaadinBootstrap.js, line 203 at r2](https://reviewable.io/reviews/vaadin/framework/10953#-LDpj8xZ6x3EwIEYgzjU:-LDqEpi4AvsYfQ7FEAn9:b-rswj4a) ([raw file](https://github.com/vaadin/framework/blob/778e751a5bfa4367aefd03fd6ce3262f3332b4d0/server/src/main/resources/VAADIN/vaadinBootstrap.js#L203)):*
<details><summary><i>Previously, elmot (Ilia Motornyi) wrote…</i></summary><blockquote>

extra spaces
</blockquote></details>

Never mind, found the issue and fixed.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/framework/10953)*
<!-- Sent from Reviewable.io -->

elmot commented:
Closed in favor of #11034
